src/main/java/org/kitteh/irc/client/library/implementation/IRCClient.java

```diff
@@ -103,9 +105,20 @@ protected void processElement(@Nonnull String element) {
     private final ActorProvider actorProvider = new ActorProvider(this);
 
     private Map<Character, ModeStatus<UserMode>> userModes;
+    private StsMachine stsMachine;
+    private boolean usingSts = false;
 
     IRCClient(@Nonnull Config config) {
         this.config = config;
+        if (this.config.get(Config.STS_STORAGE_MANAGER) != null) {
+            this.configureSts();
+        } else if (!this.config.getNotNull(Config.SSL)) {
+            throw new KittehAuthorNagException(
```

Comment:
So KICL shuts down entirely when this statement is true? I'm not sure if I'd agree with that, especially not if you're not going to reference documentation on how to easily enable STS support, because for many people this will be foreign language and they won't care without being told the benefits of it. If my software shut down on me because I was using sub-optimal settings, rather than fix it for me or throw a warning rather than shutting down. I'd probably stop using it.

If this is so important, I think this if-statement could be used to enable STS support by default (automated detection),  rather than to shut down the entire program. Perhaps with some way to indicate to the owner that this is happening; provided it has no negatives associated with it (which  it probably doesn't, if shutting down is the alternative.)